src/test/java/uk/gov/ch/developer/docs/controller/developer/SignOutControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.ch.developer.docs.controller.developer;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.ch.oauth.IIdentityProvider;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.handler.SessionHandler;
+import uk.gov.companieshouse.session.model.SignInInfo;
+import uk.gov.companieshouse.session.store.Store;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SignOutControllerTest {
+
+    @Mock
+    private IIdentityProvider identityProviders;
+    @Mock
+    private Store store;
+    @Mock
+    private HttpServletRequest httpServletRequest;
+    @Mock
+    private HttpServletResponse httpServletResponse;
+    @Mock
+    private Session session;
+    private Map<String, Object> data;
+
+    @InjectMocks
+    private SignOutController signOutController;
+
+    @Test
+    @DisplayName("Test that a valid signed in user's session state is correctly altered")
+    public void testDoSignOutWhenSignedIn() throws IOException {
+        final String zxsValue = "0000000001z";
+        SignInInfo signInInfo = new SignInInfo();
+        signInInfo.setSignedIn(true);
+
+        when(httpServletRequest.getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY))
+                .thenReturn(session);
+        when(session.getSignInInfo()).thenReturn(signInInfo);
+        when(session.getData()).thenReturn(setUserSessionData(zxsValue));
+
+        signOutController.doSignOut(httpServletResponse, httpServletRequest);
+
+        assertFalse(data.containsKey(SessionKeys.SIGN_IN_INFO.getKey()));
+
+        verify(store).delete(zxsValue);
+        verify(httpServletResponse).sendRedirect(identityProviders.getRedirectUriPage());
```

Comment:
I don't like this. 
Since identity provider is mocked I think this just evaluates to
`        verify(httpServletResponse).sendRedirect(null);`
which is very ambigous as to did the actual thing happen.
Would be good to get a local value for the return page and mock this correctly.